core/lib/instrument_api.h

```diff
@@ -6903,4 +6903,11 @@ dr_unregister_persist_patch(bool (*func_patch)(void *drcontext, void *perscxt,
                                                byte *bb_start, size_t bb_size,
                                                void *user_data));
 
+DR_API
+/**
+ * Query whether detach is in progress.
+ */
+bool
+dr_doing_detach(void);
```

Comment:
Maybe the name should have "is" to reflect the bool query and to match dr_is_notify_on, dr_is_thread_native, 
dr_is_tracking_where_am_i.  `dr_is_detaching`?

Reply:
Done.